src/us/kbase/test/auth2/MongoController.java

```diff
@@ -0,0 +1,147 @@
+package us.kbase.test.auth2;
+
+import static us.kbase.common.test.controllers.ControllerCommon.checkExe;
+import static us.kbase.common.test.controllers.ControllerCommon.findFreePort;
+import static us.kbase.common.test.controllers.ControllerCommon.makeTempDirs;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+
+import com.github.zafarkhaja.semver.Version;
+import org.apache.commons.io.FileUtils;
+
+
+/** Q&D Utility to run a Mongo server for the purposes of testing from
+ * Java.
+ * @author gaprice@lbl.gov, sijiex@lbl.gov
+ *
+ */
+public class MongoController {
+
+    private final static String DATA_DIR = "data";
+
+    private final static List<String> tempDirectories =
+            new LinkedList<String>();
+    static {
+        tempDirectories.add(DATA_DIR);
+    }
+
+    private final static Version MONGO_DB_3 =
+            Version.forIntegers(3,6,23);
```

Comment:
Was `--no-journal` removed after this version? If not the version here should be specified for when it was removed

Reply:
> In version 3.6, the [--nojournal](https://www.mongodb.com/docs/v3.6/reference/program/mongod/#cmdoption-mongod-nojournal) option is deprecated

https://www.mongodb.com/docs/v3.6/tutorial/manage-journaling/

3.6.23 is the last patch of v3.6 and --no-journal works fine with it.

Reply:
That says it's only for replica set members, which isn't applicable to this case

Reply:
Looks like it's still available in 6: https://www.mongodb.com/docs/v6.0/tutorial/manage-journaling/

Reply:
Removed in 6.1: https://www.mongodb.com/docs/manual/release-notes/6.1/#changes-to-journaling

Reply:
👍

Reply:
:+1: